### PR TITLE
Fix bug with layer caching in get_event_keycode

### LINKED ISSUE #8600: Recent commit breaks LT(5,KC_J) on FC660C
## Describe the Bug

I had fallen 27 commits behind master while working on several keyboards. After catching up to March 28 by merging upstream master into my fork, LT(5,KC_J) and similar keycodes stopped working on my FC660C (they still worked on my Planck EZ; I have a DZ60 I didn't test). Rather than momentarily activating the layer, I get stuck in that layer until I unplug and reconnect the keyboard.

While exploring this bug, I added an emergency escape DF(1) from every layer back to my base layer. These keys failed to work at all; I remained stuck in the 'momentary' layer.

Reverting the 27 upstream commits fixed the bug.

My keyboard is [fc660c syzygies](https://github.com/Syzygies/qmk_firmware/tree/syzygies/keyboards/fc660c/keymaps/syzygies)

My master branch was caught up with the upstream master as of yesterday, and causes the problem. My syzygies branch is my development branch, that again works once I reverted it to discard these 27 commits. 

My fork's git history should be public, making precise which commits I've fallen behind on.

## System Information

 - Keyboard:  FC660C with a Hasu controller
   - Revision (if applicable):  
 - Operating system: MacOS 10.14.6
 - AVR GCC version: avr-gcc (GCC) 8.3.0
 - ARM GCC version: 
arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 8-2019-q3-update) 8.3.1 20190703 (release) [gcc-8-branch revision 273027]
 - QMK Firmware version:
0.8.79 (both branches report this same version)
 - Any keyboard related software installed? 
   - [ ] AutoHotKey
   - [ ] Karabiner
   - [x] Other: Alfred 4.0.9 [1144]

## Additional Context

On request I can do a binary search to determine which of the 27 commits is responsible, and I can see if the DZ60 is affected. Usually I find my less obvious bugs by waiting a few hours and seeing if I slap my forehead. If someone's forehead slap doesn't resolve the bug, but suggests a commit to look at first, that would save me time.

I am a Kinesis Advantage survivor, so of course the first thing I considered was dropped USB events, but that wouldn't explain the escape DF(1) not working, or reverting the 27 upstream commits fixing everything. This is not sporadic behavior, it is 100% reproducible.

### PR DESCRIPTION
**Pull request by drashna**
## Description

Basically, the `get_event_keycode` handler was updating the layer cache when it shouldn't have been.  This was causing Layer Tap to get stuck, since it was no longer valid.

This adds an additional check to ensure if the layer cache should be updated or not. And updates all of the functions that are calling it (to be "false", so they don't update the layer cache, and instead read the layer cache.

## Types of Changes
- [x] Core
- [x] Bugfix

## Issues Fixed or Closed by This PR

* Fixes #8600

## Checklist
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/common/action.c [hunk 1]
```diff
@@ -188,7 +188,7 @@ void process_record(keyrecord_t *record) {
     if (!process_record_quantum(record)) return;
 
     process_record_handler(record);
-    post_process_record_quantum(record);
+    //post_process_record_quantum(record);
```
Comment (tzarc) [inline tmk_core/common/action.c:191]: This intentional?

File: tmk_core/common/action.c [hunk 2]
```diff
@@ -324,7 +324,7 @@ void process_action(keyrecord_t *record, action_t action) {
 #    if !defined(IGNORE_MOD_TAP_INTERRUPT) || defined(IGNORE_MOD_TAP_INTERRUPT_PER_KEY)
                             if (
 #        ifdef IGNORE_MOD_TAP_INTERRUPT_PER_KEY
-                                !get_ignore_mod_tap_interrupt(get_event_keycode(record->event)) &&
+                                !get_ignore_mod_tap_interrupt(get_event_keycode(record->event,false)) &&
```
Comment (tzarc) [inline tmk_core/common/action.c:327]:
```suggestion
                                !get_ignore_mod_tap_interrupt(get_event_keycode(record->event, false)) &&
```

File: tmk_core/common/action.c (hunk 1 above)
Comment (drashna) [reply tmk_core/common/action.c:191]: Nope

**Comment by drashna** [comment]
> Are there any docs that need updating, as well?

No, it shouldn't.  

There are a number of per-key features that use this to get the keycode.  That's why the "bool" was added and the check for it.  It's only "true" and updates the layer cache if it's actually being called from `process_action`, and never anywhere else.  

As for needing documentation.... possibly, but not in an existing docs. However, I should probably document why this is the case in the code.

**Comment by tzarc** [comment]
Could we move the main implementation (i.e. with true) to be a "hidden" api only callable from `process_action`? Just worried people will use this with true without understanding the implications, and in the future if someone creates a new feature and doesn't understand the semantics of the API it'll end up requiring a whole bunch of investigation.

**Comment by drashna** [comment]
in that case, I'll add a note in the code, and change the bool to `update_layer_cache` to make it more clear what is going on (naming, ugh)

**Comment by EricGebhart** [comment]
Tested on xd75:ericgebhart, It is the same code I have for ergodox_ez:ericgebhart. Everything works as expected.   Thank you.

**Pull request merged**